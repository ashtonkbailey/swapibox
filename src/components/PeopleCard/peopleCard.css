article {
  color:  #FBF230;
  font-size: 16px;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 6px;
  transition: background-color, opacity .6s ease-in-out;
}

article>img {
  margin: 1em;
}

.saber {
  height: 60px;
  width: 18%;
  position: relative;
  bottom: 16px;
  margin-top: 10px;
}

.saber:hover {
  /* opacity: 30; */
  cursor: pointer;
  transition: .3s ease-in-out;
  transform: scale(1.1);
}

.name {
  margin: 0;
  font-family: Roboto;
  font-style: normal;
  font-size: 40px;
  text-align: center;
  color: black;
  text-shadow:
  -1px -1px 0 rgb(238, 213, 19),  
  1px -1px 0 rgb(238, 213, 19),
  -1px 1px 0 rgb(238, 213, 19),
  1px 1px 0 rgb(238, 213, 19);
}

.small-card .name {
  font-size:  20px; 
}

.small-card {
  height: 500px;
  transition: background-color, opacity .6s ease-in-out;
  background: rgba(227, 218, 135, 0.2);
  /* max-width: 300px; */
  /* width: 80%; */
  /* padding-top: 20px; */
  margin-top: 60px;
  pointer-events: none;
}

.small-card>.saber {
  width: 60px;
}

.left-card {
  /* grid-column: 1/2; */
  opacity: .5;
  grid-row: 1;
  position: fixed;
  /* bottom: 210px; */
  width: 22%;
  left: 5%;
}

.right-card {
  grid-column: 3/4;
  opacity: .5;
  /* grid-row: 1; */
  position: fixed;
  width: 22%;
  /* bottom: 210px; */
  right: 5%;
}


.middle-card {
  /* grid-column: 2/3; */
  position: fixed;
  left: 32%;
  font-size: 24px;
  /* font-family: Verdana, Geneva, Tahoma, sans-serif; */
  /* min-width: 300px; */
  width: 34%;
  height: 500px;
  background: rgba(227, 218, 135, 0.4);
  border: 4px solid #EDE748;
  /* box-sizing: border-box; */
  box-shadow: 0px 4px 40px rgb(227, 218, 135);
  border-radius: 40px;
  margin: 20px;
  /* padding: 5px; */
  /* grid-row: 1; */
}

.small-card>img {
  width: 32%;
}

.middle-card>.saber {
  width: 16%;
  height: 16%;
}

.luke {
  border: solid 1px black;
  border-radius: 20px;
  box-shadow: 2px 2px  black;
  margin: 2px;
}

.left-card, .right-card {
  height: 400px;
  background: rgba(0, 0, 0, 0.66);
  border: 2px solid #EDE748;
  border-radius: 40px;
}

p {
  margin: 10px;
  color: black;
  text-shadow: 
  .5px .5px 0 rgb(238, 213, 19);
}
