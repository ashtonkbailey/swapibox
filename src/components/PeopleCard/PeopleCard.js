import React from 'react';
import PropTypes from 'prop-types';

import lightsabers from '../../images/lightsabers.png';
import './peopleCard.css';
import luke from '../../images/luke.jpg';

const PeopleCard = ({
  card,
  carouselIndex,
  addToFavorites,
  chosenContent
}) => {
  let hiddenClass;
  const leftCardIndex = carouselIndex;
  let middleCardIndex = carouselIndex + 1;
  let rightCardIndex = carouselIndex + 2;

  if (carouselIndex === 9) {
    middleCardIndex = 0;
    rightCardIndex = carouselIndex - 8;
  }

  if (carouselIndex === 8) {
    rightCardIndex = carouselIndex - 8;
  }

  switch (card.index) {
    case middleCardIndex:
      hiddenClass = 'middle-card';
      break;
    case leftCardIndex:
      hiddenClass = 'left-card small-card';
      break;
    case rightCardIndex:
      hiddenClass = 'right-card small-card';
      break;
    default:
      hiddenClass = 'hidden';
  }
  
  if (chosenContent === 'people') {
    return (
      <article className={hiddenClass}>
        <h3 className="name">
          {card.name}
        </h3>
        <img
          src={luke}
          alt="character"
        />
        <p>
          species:
          {card.species}
        </p>
        <p>
          homeworld:
          {card.homeworld}
        </p>
        <p>
          population:
          {card.population}
        </p>
        <img
          className={ card.favorite ? "saber active" : "saber" }
          src={lightsabers}
          onClick={() => addToFavorites({ card })}
          alt="click to add to favorites"
        />
      </article>
    );
  }
  else if (chosenContent === 'planets') {
    return (
      <article className={hiddenClass}>
        <h3 className="name">
          {card.name}
        </h3>
        <img
          src={luke}
          className="luke"
          alt="planet"
        />
        <p>
          terrain:
          {card.terrain}
        </p>
        <p>
          climate:
          {card.climate}
        </p>
        <p>
          population:
          {card.population}
        </p>
        <p>
          residents:
          {card.residents}
        </p>
        <img
          className={ card.favorite ? "saber active" : "saber" }
          src={lightsabers}
          onClick={() => addToFavorites({ card })}
          alt="click to add to favorites"
        />
      </article>
    );
  }
    else if (chosenContent === 'vehicles') {
      return (
        <article className={hiddenClass}>
          <h3 className="name">
            {card.name}
          </h3>
          <img
            src={luke}
            alt="vehicle"
          />
          <p>
            model:
            {card.model}
          </p>
          <p>
            class:
            {card.class}
          </p>
          <p>
            passengers:
            {card.passengers}
          </p>
          <img
            className={ card.favorite ? "saber active" : "saber" }
            src={lightsabers}
            onClick={() => addToFavorites({ card })}
            alt="click to add to favorites"
          />
        </article>
      );
  }
};

PeopleCard.propTypes = {
  card: PropTypes.object.isRequired,
  carouselIndex: PropTypes.func,
  addToFavorites: PropTypes.func,
  chosenContent: PropTypes.array.isRequired,
};

export default PeopleCard;
