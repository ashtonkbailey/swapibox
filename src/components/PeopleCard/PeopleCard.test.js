import React from 'react';
import { shallow } from 'enzyme';

import PeopleCard from './PeopleCard';

describe('people card', () => {
  let mockCard;
  let wrapper;
  let mockAddToFavorites;

  beforeEach(() => {
    mockCard = {
      card: {
        favorite: false,
        homeworld: 'Tatooine',
        index: 0,
        name: 'Luke Skywalker',
        population: '200000',
        species: 'Human',
      },
    };
    mockAddToFavorites = jest.fn();
    wrapper = shallow(<PeopleCard card={mockCard.card} addToFavorites={mockAddToFavorites} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('calls add to favorites method with the correct parameters', () => {
    const button = wrapper.find('.saber');
    button.simulate('click');
    expect(mockAddToFavorites).toHaveBeenCalledWith(mockCard);
  });
});
