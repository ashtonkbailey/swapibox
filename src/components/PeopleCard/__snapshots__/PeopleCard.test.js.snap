// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`people card should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PeopleCard
    addToFavorites={[MockFunction]}
    card={
      Object {
        "favorite": false,
        "homeworld": "Tatooine",
        "index": 0,
        "name": "Luke Skywalker",
        "population": "200000",
        "species": "Human",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): undefined,
  Symbol(enzyme.__nodes__): Array [
    undefined,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
